### ISSUE TITLE
Issues with command line switches

### ISSUE THREAD

**Issue by ArkieCoder** [ntreeinc/apex-source-control#2]
Hi guys,

Great work on this!  I am forced against my will and better judgement to use APEX, and things like this really help bring some sanity to my world.

Two things I had to do to make this thing export correctly:

- Remove the '-s' command line switch for the 'which' command.  
- Remove the space between '-i' and 'bak' in the sed command in apexupdate.sh

After that, things worked just fine.  I'm running the command from within a Docker container.  Here's my Dockerfile:
```
FROM node:current
## add contrib to sources.list for make-*-package
RUN perl -p -i -e 's/.*/$& contrib/' /etc/apt/sources.list
RUN apt-get -yqq update
## dependencies for make-*-package
RUN apt-get -y install openjdk-8-jdk build-essential binutils-multiarch dh-exec devscripts fakeroot tofrodos unzip xterm
## create sqldeveloper package and install
RUN wget https://github.com/lazarusllong/sqldeveloper-package/releases/download/0.5.4/sqldeveloper-package_0.5.4_all.deb
RUN dpkg -i sqldeveloper-package_0.5.4_all.deb
COPY sqldeveloper-18.2.0.183.1748-no-jre.zip .
RUN make-sqldeveloper-package sqldeveloper-18.2.0.183.1748-no-jre.zip
RUN bash -c "dpkg -i sqldeveloper-[0-9]*.deb"
RUN bash -c "dpkg -i libjnidispatch*.deb"
RUN ln -s /usr/bin/sql.18.2.0.183.1748 /usr/bin/sql
## set timezone to avoid ORA-01882 error
ENV TZ=America/Chicago
WORKDIR /tmp/output
CMD sleep infinity
```

... and my docker-compose.yml:

```
version: '3'
services:
  node:
    build: .
    volumes:
      - .:/tmp/output
```

I will modify this in the future so that I can just supply command line arguments to a script which will run the docker container and give me the output.

If you can help me understand if there's a different version of 'which' or 'sed' which I should be using so I don't have to modify your code, I'd be grateful.  Thanks!

**Comment by michaelabuckley** [comment]
Happy to hear that this is useful for you.  We're running this on OSX which has BSD tools, so "which -s" is silent, and it seems that sed has slightly different behaviour.  Go ahead hand submit a patch with a fix for sed, and replace "which -s" with (which > /dev/null).  I'm unlikely to work on this any time soon since we only deploy on linux.

Honestly, I should just merge this all into one bash script with sub-commands for simplicity.

Also FYI: much of this becomes redundant with Oracle's new sqlcl tool.  It has apex export built in.

**Comment by ArkieCoder** [comment]
This patch worked for the Docker container I was creating
```
diff -uNr node_modules.orig/apex-source-control/scripts/apexupdate.sh node_modules/apex-source-control/scripts/apexupdate.sh
--- node_modules.orig/apex-source-control/scripts/apexupdate.sh	2019-03-15 14:35:10.268801060-0500
+++ node_modules/apex-source-control/scripts/apexupdate.sh	2019-03-15 14:40:28.927551508 -0500
@@ -55,7 +55,7 @@
 }
 
 function main {
-    if which -s sql; then
+    if which sql; then
 	source ./config/asc.conf
 	local connect_string="${username}/${password}@${database_connection}"
 	echo Using sqlcl to export app ${apexappid}
@@ -82,7 +82,7 @@
     mv "f${apexappid}" apex
 
     # fixup install paths.  The tool generates absolute paths
-    sed -E -i 'bak' 's^@(.*/)?application/^@apex/application/^' apex/install.sq
l
+    sed -E -i'bak' 's^@(.*/)?application/^@apex/application/^' apex/install.sql
 }
 
 main
diff -uNr node_modules.orig/apex-source-control/scripts/file-to-apex node_modules/apex-source-control/scripts/file-to-apex
--- node_modules.orig/apex-source-control/scripts/file-to-apex	2019-03-15 14:35:10.268801060 -0500
+++ node_modules/apex-source-control/scripts/file-to-apex	2019-03-15 14:40:05.207604408 -0500
@@ -5,7 +5,7 @@
 NLS_LANG=.AL32UTF8
 export NLS_LANG
 
-if which -s sql; then
+if which sql; then
     sql_cmd=sql
 else
     echo WARNING: could not find SQLcl \(sql\).  Falling back to sqlplus
diff -uNr node_modules.orig/apex-source-control/scripts/generate-app-id node_modules/apex-source-control/scripts/generate-app-id
--- node_modules.orig/apex-source-control/scripts/generate-app-id	2019-03-15 14:35:10.268801060 -0500
+++ node_modules/apex-source-control/scripts/generate-app-id	2019-03-15 14:39:49.155643225 -0500
@@ -2,7 +2,7 @@
 script_folder=$(dirname $0)
 source config/asc.conf || exit 1
 
-if which -s sql; then
+if which sql; then
     echo Running PWD $PWD
     sql_cmd="sql -S"
 else
diff -uNr node_modules.orig/apex-source-control/scripts/uninstall-apex node_modules/apex-source-control/scripts/uninstall-apex
--- node_modules.orig/apex-source-control/scripts/uninstall-apex	2019-03-15 14:35:10.268801060 -0500
+++ node_modules/apex-source-control/scripts/uninstall-apex	2019-03-15 14:39:14.803734897 -0500
@@ -3,7 +3,7 @@
 "${script_folder}"/check_conf_file.sh || exit 1
 source config/asc.conf || exit 1
 
-if which -s sql; then
+if which sql; then
     sql_cmd=sql
 else
     echo WARNING: could not find SQLcl \(sql\).  Falling back to sqlplus
```
I will share my Dockerfile and docker-compose.yml file with you if you are interested.  My goal was to be able to quickly export an APEX application with a single command line but without having to have all the prerequisites installed on my machine.

That is interesting about sqlcl.  I will definitely look into that to see if it's possible to containerize it as well.

**Comment by michaelabuckley** [comment]
Fixed in 1.3
